uint64 reasoner_id
uint8 idle = 0
uint8 reasoning = 1
uint8 executing = 2
uint8 finished = 3
uint8 inconsistent = 4
uint8 deliberative_state